File: src/JavaScript/easy/string/problem171.js
/**
 * @param {string} columnTitle
 * @return {number}
 */

var titleToNumber = function(columnTitle) {
    let sum = 0
    for(let j = 0; j < columnTitle.length; j++){
        sum += (columnTitle.charCodeAt(j) - 64) * Math.pow(26, columnTitle.length - j - 1)
    }
    return sum
};

// let columnTitle = "A"
// let columnTitle = "AB"
// let columnTitle = "ZY"
// let columnTitle = "FXS"
let columnTitle = "FXSHRXW"
// 2147483647

titleToNumber(columnTitle)

File: src/Python/easy/string/problem171.py
import math

class Solution:
    def titleToNumber(self, columnTitle: str) -> int:
        sum = 0
        for i in range(len(columnTitle)):
            sum += int((ord(columnTitle[i]) - 64) * math.pow(26, len(columnTitle) - i - 1))
        return sum



# columnTitle = "AB"
columnTitle = "ZY"

solution = Solution()
solution.titleToNumber(columnTitle)
